go/mcap/reader.go

```diff
@@ -141,29 +141,33 @@ func (r *Reader) Messages(
 	return r.unindexedIterator(ro.Topics, uint64(ro.Start), uint64(ro.End)), nil
 }
 
-func (r *Reader) readHeader() (*Header, error) {
-	_, err := r.rs.Seek(8, io.SeekStart)
-	if err != nil {
-		return nil, fmt.Errorf("failed to seek to header: %w", err)
+// Header reads the Header record from the MCAP.
+// NOTE: on non-seeking readers, this method will return an error if any other records
+// have been read from the MCAP file before.
+func (r *Reader) Header() (*Header, error) {
+	var err error
+	if r.rs != nil {
+		_, err = r.rs.Seek(0, io.SeekStart)
+		if err != nil {
+			return nil, fmt.Errorf("failed to seek to start: %w", err)
+		}
+		r.l, err = NewLexer(r.r, &LexerOptions{EmitChunks: true})
```

Comment:
is it necessary to make a new lexer here, vs seeking the reader that underlies the one that is attached to the reader already?